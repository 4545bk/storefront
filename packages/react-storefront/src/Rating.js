/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import Star from '@material-ui/icons/Star'
import StarBorder from '@material-ui/icons/StarBorder'
import StarHalf from '@material-ui/icons/StarHalf'
import withStyles from '@material-ui/core/styles/withStyles'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Hbox } from './Box'

/**
 * Displays a star rating corresponding to the provided value
 */
export const styles = theme => ({
  root: {
    display: 'flex',
    '& svg': {
      color: theme.palette.rating,
      height: '16px',
      width: '16px',
      marginLeft: '-2px',
      display: 'block'
    }
  },
  blank: {
    '& svg': {
      color: theme.palette.divider
    }
  },
  reviewsLabel: {
    marginLeft: '3px'
  }
});

@withStyles(styles, { name: 'RSFRating' })
export default class Rating extends Component {

  static propTypes = {
    /**
     * CSS classes to apply
     */
    classes: PropTypes.object,

    /**
     * The number of stars to display.  Can be an integer or a float
     * ending in .5.
     */
    value: PropTypes.number,

    /**
     * The number of reviews
     */
    reviewCount: PropTypes.number,

    /**
     * A function that returns the label displayed to the right of the review count.
     * For example: `<Rating label={reviewCount => <span> {reviewCount == 1 ? 'review' : 'reviews'}</span>}/>`.  
     * This value used in this example is the default.
     */
    label: PropTypes.func,

    /**
     * An instance of ProductModelBase.  You can set this instead of setting value an reviewCount individually.
     */
    product: PropTypes.object
  }

  static defaultProps = {
    label: reviewCount => <span> {reviewCount == 1 ? 'review' : 'reviews'}</span>
  }

  render() {
    let { classes, value, label, reviewCount, className, product } = this.props
    let stars = []

    if (product) {
      reviewCount = product.reviewCount
      value = product.rating
    }
  
    for (let i=1; i<=5; i++) {
      if (value == null || value >= i) {
        stars.push(<Star key={i}/>)
      } else if (value >= i - 0.5) {
        stars.push(<StarHalf key={i}/>)
      } else {
        stars.push(<StarBorder key={i}/>)
      }
    }
  
    return (
      <Hbox>
        <div className={classnames(classes.root, className, { [classes.blank]: value == null })}>
          {stars}
        </div>
        { reviewCount ? (
          <div className={classes.reviewsLabel}>({reviewCount}{label(reviewCount)})</div>
        ) : null}
      </Hbox>
    )
  }
}
