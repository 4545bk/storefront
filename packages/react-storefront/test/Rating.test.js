/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Rating from '../src/Rating'

describe('Rating', () => {

  it('renders 0', () => {
    const component = <Rating value={0}/>
    expect(mount(component)).toMatchSnapshot()
  })

  it('renders 5', () => {
    const component = <Rating value={5}/>
    expect(mount(component)).toMatchSnapshot()
  })

  it('should display half stars', () => {
    const component = <Rating value={3.5}/>
    expect(mount(component)).toMatchSnapshot()
  })

  it('renders null', () => {
    const component = <Rating value={null}/>
    expect(mount(component)).toMatchSnapshot()
  })

  it('accepts a custom label', () => {
    const component = (
      <Rating 
        value={2} 
        reviewCount={10} 
        label={count => (<span>customers reviewed this product</span>)}
      />
    )

    expect(mount(component)).toMatchSnapshot()
  })
})