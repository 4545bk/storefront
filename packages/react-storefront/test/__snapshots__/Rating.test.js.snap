// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating accepts a custom label 1`] = `
<WithStyles(Rating) value={2} reviewCount={10} label={[Function: label]}>
  <Rating classes={{...}} value={2} reviewCount={10} label={[Function: label]}>
    <Hbox>
      <WithStyles(Box) alignItems="center" flexDirection="row">
        <Box classes={{...}} alignItems="center" flexDirection="row">
          <div className="RSFBox-root-4" style={{...}}>
            <div className="RSFRating-root-1">
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
            </div>
            <div className="RSFRating-reviewsLabel-3">
              (
              10
              <span>
                customers reviewed this product
              </span>
              )
            </div>
          </div>
        </Box>
      </WithStyles(Box)>
    </Hbox>
  </Rating>
</WithStyles(Rating)>
`;

exports[`Rating renders 0 1`] = `
<WithStyles(Rating) value={0}>
  <Rating classes={{...}} value={0} label={[Function: label]}>
    <Hbox>
      <WithStyles(Box) alignItems="center" flexDirection="row">
        <Box classes={{...}} alignItems="center" flexDirection="row">
          <div className="RSFBox-root-4" style={{...}}>
            <div className="RSFRating-root-1">
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
            </div>
          </div>
        </Box>
      </WithStyles(Box)>
    </Hbox>
  </Rating>
</WithStyles(Rating)>
`;

exports[`Rating renders 5 1`] = `
<WithStyles(Rating) value={5}>
  <Rating classes={{...}} value={5} label={[Function: label]}>
    <Hbox>
      <WithStyles(Box) alignItems="center" flexDirection="row">
        <Box classes={{...}} alignItems="center" flexDirection="row">
          <div className="RSFBox-root-4" style={{...}}>
            <div className="RSFRating-root-1">
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
            </div>
          </div>
        </Box>
      </WithStyles(Box)>
    </Hbox>
  </Rating>
</WithStyles(Rating)>
`;

exports[`Rating renders null 1`] = `
<WithStyles(Rating) value={{...}}>
  <Rating classes={{...}} value={{...}} label={[Function: label]}>
    <Hbox>
      <WithStyles(Box) alignItems="center" flexDirection="row">
        <Box classes={{...}} alignItems="center" flexDirection="row">
          <div className="RSFBox-root-4" style={{...}}>
            <div className="RSFRating-root-1 RSFRating-blank-2">
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
            </div>
          </div>
        </Box>
      </WithStyles(Box)>
    </Hbox>
  </Rating>
</WithStyles(Rating)>
`;

exports[`Rating should display half stars 1`] = `
<WithStyles(Rating) value={3.5}>
  <Rating classes={{...}} value={3.5} label={[Function: label]}>
    <Hbox>
      <WithStyles(Box) alignItems="center" flexDirection="row">
        <Box classes={{...}} alignItems="center" flexDirection="row">
          <div className="RSFBox-root-4" style={{...}}>
            <div className="RSFRating-root-1">
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(Star)>
                <Star>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </Star>
              </pure(Star)>
              <pure(StarHalf)>
                <StarHalf>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4V6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarHalf>
              </pure(StarHalf)>
              <pure(StarBorder)>
                <StarBorder>
                  <WithStyles(SvgIcon)>
                    <SvgIcon classes={{...}} color="inherit" viewBox="0 0 24 24">
                      <svg className="MuiSvgIcon-root-6" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true">
                        <g>
                          <path d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z" />
                        </g>
                      </svg>
                    </SvgIcon>
                  </WithStyles(SvgIcon)>
                </StarBorder>
              </pure(StarBorder)>
            </div>
          </div>
        </Box>
      </WithStyles(Box)>
    </Hbox>
  </Rating>
</WithStyles(Rating)>
`;
